test/jdk/com/sun/jdi/VMConnection.java

```diff
@@ -51,7 +51,10 @@ static public String getDebuggeeVMOptions() {
 
         // When we run under jtreg, test.classes contains the pathname of
         // the dir in which the .class files will be placed.
-        String testClasses = System.getProperty("test.classes");
+        // When we run jtreg with plugin, the TestScaffold should be also in classpath
```

Comment:
A stupid question about this part of comment: `the TestScaffold should be also in classpath`.
If it should be ALSO in classpath then why in the `Virtual` case we use "test.class.path" instead of "test.classes" but not both. It'd be good to clarify this in the comment.

Reply:
The "test.class.path" is complete classpath required to run test. It included test.classes as well as testlibrary and some other classes. The tests located in com/sun/jdi/<subdirectory> uses 'com/sun/jdi' as a testlibrary.
The better comment would be
// When we run jtreg with plugin, the full classpath including testlibrary is required to use TestScaffold is required

However, before fixing this comment I would like to see if there are any objections against using "test.class.path"  in all cases. It would minimize difference in virtual and standard mode. 
Is there are any reason to use "test.classes" at all?

Reply:
I'd prefer to always use the "test.class.path" as it is a complete classpath required to run test.

> The better comment would be
> // When we run jtreg with plugin, the full classpath including testlibrary is required to use TestScaffold is required

I guess, you are going to remove this duplication with "is required". :)

Reply:
I think the CR should explain this test.classes vs test.class.path difference, and also explain why the classpath requirements are different when using virtual threads. You say that "TestScaffold should be also in classpath", but don't explain why. It's also not clear to me why only tests that are in subdirs are impacted when running with the wrapper.

Also, when you say "When we run jtreg with plugin..", do you mean with the wrapper, not the plugin?

Reply:
@plummercj, I'e updated CR with more detailed information. Hope it makes clearer why this change is needed. Here is the text.

"
VMConnection runs debuggee using "test.classes" as a classpath for debuggee classes. It works fine when test and TestScaffold.java are located in the same directory and are both compiled into "test.classes" location. However, it causes test failures when the virtual thread factory (wrapper) is enabled and the test is not located in the same directory as TestScaffold. Such tests use TestScaffold as a testlibrary. (They have ' * @library ..' or something like similar).
So TestScaffold is not included in 'test.classes' which include tests only. However, it is included as all testlibrary classes in "test.class.path" classparth.

It is needed to use "test.class.path" to be able to use TestScaffolld as debugee wrapper Might be it better to always use "test.class.path" as classpath for debugee to minimize the difference between standard and "virtual" execution.
"

Yes, it would be better to change "When we jtreg with plugin" to "When 'main wrapper' is used". 

Before updating the comments I want to confirm if we want to keep "test.classes" for standard execution or change it to use  "test.class.path" always.

Reply:
I'm ok with always using test.class.path.